src/utils.ts

```diff
@@ -0,0 +1,195 @@
+export function oneLineStringComponentsToArray(line: string) {
+  const delimiters = ['[', ']', '{', '}', '(', ')', ',', '.', '=', '>', '<', '*', '/', '%', '&', '|', '!', ':', '?'];
+  const doubleDelimiters = ['||', '&&', '==', '!=', '<=', '>=', '=>', '${', '%{', '..'];
+  // the .. is the initial of the triple delimiter ...
+
+  const charsArray = line.trim().split('');
+  let inString = false;
+  let ignoreNextChar = false;
+  const arrayComponents = [] as { separator: string; value: string }[];
+  let currentComponentCharArray: string[] = [];
+  function addCurrentAndPrepareNext() {
+    if (currentComponentCharArray.length) {
+      arrayComponents.push({ separator: lastSeparator, value: currentComponentCharArray.join('') });
+      currentComponentCharArray = [];
+    }
+  }
+  let lastSeparator = '';
+  for (let i = 0; i < charsArray.length; i++) {
+    const lastChar = i === 0 ? '' : charsArray[i - 1];
+    const currentChar = charsArray[i];
+    const nextChar = charsArray[i + 1] || '';
```

Comment:
Does this line will return empty string when there's no (i+1) position in the array ?

Reply:
This will return undefined when it's last element, what causes the OR expression to return the other value.